proposals/collection-literals.md

```diff
@@ -151,70 +156,72 @@ _Open issue: Relying on a parameter named `capacity` seems brittle. Is there an
+  * `Ei` is a *spread element* `Si` and there is an implicit conversion from the *iteration type* of `Si` to `T`.
 
-* the `Construct` method is found on an instance of `T` (including through [extension methods](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md#11783-extension-method-invocations)), and
-
-* `CollectionType` is some other type known to be a [*constructible*](constructible-collection-types) type.
-
-If found, the collection can be constructed by creating a fresh instance of its type using `new T()`, producing the corresponding argument to pass to `Construct`, and then calling that method on the fresh instance.  `new T()` supports all structs, including those without a `parameterless struct constructor`.
-
-The allowance for extension methods means that collection literal support can be added to an existing API which does not already directly support this.
-
-The `Construct` method is used for construction of the collection even if the type supports *collection initializers*.
-_This means an extension method could be added that would silently change how collection literals for an existing collection initializer type are constructed in the program._
+Types for which there is an implicit collection literal conversion from a collection literal are the valid *target types* for that collection literal.
 
-Through the use of the [`init`](#init-methods) modifier, existing APIs can directly support collection literals in a manner that allows for no-overhead production of the data the final collection will store.
+## Create methods
+[create-methods]: #create-methods
 
-_Does this support construction of custom dictionary types?_
+A *create method* is indicated with a `[CollectionBuilder]` attribute on the *collection type*.
+The attribute specifies the *builder type* and *method name* of a method to be invoked to construct an instance of the collection type.
 
-### `init Construct` methods
-[init-methods]: #init-methods
+```c#
+namespace System.Runtime.CompilerServices
+{
+    public sealed class CollectionBuilderAttribute : System.Attribute
+    {
+        public CollectionBuilderAttribute(Type builderType, string methodName);
+        public Type BuilderType { get; }
+        public string MethodName { get; }
+    }
+}
+```
 
-* Like [*`init` accessors*](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/init.md#init-only-setters), an `init` method would be an instance method invocable at the point of object creation but become unavailable once object creation has completed. This facility thus prevents general use of such a marked method outside of known safe compiler scopes where the instance value being constructed cannot be observed until complete.
+The attribute can be applied to `class` and `struct` types only.
```

Comment:
seems like it would be fine to be on an interface type.  indeed, you could imagine adding this to [`IImutableSet<T>`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.immutable.iimmutableset-1?view=net-7.0)